roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -0,0 +1,104 @@
+    tags = ["preflight"]
+
+    # attempt to pull required docker images and fail if
+    # any images are missing, or error occurs during pull
+    def run(self, tmp, task_vars):
+        is_containerized = get_var(task_vars, "openshift", "common", "is_containerized")
+        openshift_release = get_var(task_vars, "openshift_release")
+        openshift_image_tag = get_var(task_vars, "openshift_image_tag")
+        deployment_type = get_var(task_vars, "deployment_type")
+
+        is_origin_deployment = deployment_type == "origin"
+
+        changed = False
+        failed_pulls = set()
+
+        required_images = self.rpm_docker_images(is_origin_deployment)
+        if is_containerized:
+            required_images = self.containerized_docker_images(is_origin_deployment)
+
+        pulled_images = self.pull_images(self.add_image_tags(required_images, openshift_release), tmp, task_vars)
+
+        if "failed_images" in pulled_images:
+            failed_pulls.update(pulled_images["failed_images"])
+
+        if not openshift_image_tag:
+            # if we did not fail to pull at least one image, then
+            # we have successfully changed the host environment
+            if len(failed_pulls) < len(required_images):
+                changed = True
+            return {"changed": changed, "failed": True, "msg": "Unable to retrieve fully qualified image tag. Failed to fetch images: %s" % (set(failed_pulls).update(self.qualified_docker_images(is_origin_deployment)))}
+
+        # pull each required image that requires a qualified image
+        # tag (v0.0.0.0) rather than a standard release format tag (v0.0).
+        # We want to do this check for both containerized and
+        # non-containerized installations.
+        pulled_qualified_images = self.pull_images(self.add_image_tags(self.qualified_docker_images(is_origin_deployment), "v" + openshift_image_tag), tmp, task_vars)
+        if "failed_images" in pulled_qualified_images:
+            failed_pulls.update(pulled_qualified_images["failed_images"])
+
+        if failed_pulls:
+            if len(failed_pulls) < len(required_images) + len(self.qualified_docker_images(is_origin_deployment)):
+                changed = True
+            return {"changed": changed, "failed": True, "msg": "Failed to pull required docker images: %s" % failed_pulls}
+
+        return {"changed": True}
+
+    @staticmethod
+    def containerized_docker_images(is_origin_deployment):
+        if is_origin_deployment:
+            return [
+                "openshift/origin"
+            ]
+        return [
+            "openshift3/ose"
+        ]
+
+    @staticmethod
+    def rpm_docker_images(is_origin_deployment):
+        return [
+            "openshift3/registry-console",
+        ]
+
+    @staticmethod
+    def qualified_docker_images(is_origin_deployment):
+        if is_origin_deployment:
+            return [
+                "openshift/origin-haproxy-router"
```

Comment:
In some places in origin and openshift-ansible we derive image names from a pattern like `prefix-component:version`.
Example:
https://github.com/rhcarvalho/origin/blob/bd3e362de2a9f849bd2d7d3ff2a372746f35fb2a/pkg/cmd/util/variable/imagetemplate.go#L30

In Origin, the `DefaultImagePrefix` is set to `openshift/origin`, and for OCP it is `openshift3/ose`. I think it would be natural to follow that pattern, and it would be easier to maintain these lists, since we know the image names follow a pattern.